homeassistant/components/yale_smart_alarm/coordinator.py

```diff
@@ -0,0 +1,138 @@
+"""DataUpdateCoordinator for the Yale integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from yalesmartalarmclient.client import AuthenticationError, YaleSmartAlarmClient
+
+from homeassistant.config_entries import SOURCE_REAUTH, ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, LOGGER
+
+
+class YaleDataUpdateCoordinator(DataUpdateCoordinator):
+    """A Yale Data Update Coordinator."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize the Yale hub."""
+        self.entry = entry
+        self.yale: YaleSmartAlarmClient | None = None
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=DEFAULT_SCAN_INTERVAL),
+        )
+
+    async def _async_update_data(self) -> dict:
+        """Fetch data from Yale."""
+
+        updates = await self.hass.async_add_executor_job(self.get_updates)
+
+        locks = []
+        door_windows = []
+
+        for device in updates["cycle"]["data"]["device_status"]:
+            state = device["status1"]
+            if device["type"] == "device_type.door_lock":
+                lock_status_str = device["minigw_lock_status"]
+                lock_status = int(str(lock_status_str or 0), 16)
+                closed = (lock_status & 16) == 16
+                locked = (lock_status & 1) == 1
+                if not lock_status and "device_status.lock" in state:
+                    device["_state"] = "locked"
+                    locks.append(device)
+                    continue
+                if not lock_status and "device_status.unlock" in state:
+                    device["_state"] = "unlocked"
+                    locks.append(device)
+                    continue
+                if (
+                    lock_status is True
```

Comment:
I'm assuming this is what is meant? `lock_status` is an integer, so it's not the same to compare it to a boolean with `ìs`.
```suggestion
                    lock_status
```

Reply:
Sometimes things goes a bit too fast. Appreciate the feedback. Think all is ok now